design/pipeline-artifacts.md

```diff
@@ -16,8 +18,8 @@ Artifacts are distinct from other `resources` types, including `containers`, `re
 
 ```yaml
 - upload: string # a minimatch path or list of [minimatch paths](tasks/file-matching-patterns.md) to upload
-  artifact: string # identifier for this artifact (no spaces allowed), defaults to 'default'
-  prependDirectory: string # a directory path that will be prepended to all uploaded files
+  artifact: string # identifier for this artifact (no spaces allowed), defaults to the job name
+  prependPath: string # a directory path that will be prepended to all uploaded files
```

Comment:
do we need both `prependPath` for upload and `path` for download?
What value `prependPath` can give us, since customer can use `path` to set download path to whatever they want?

Reply:
prepend path in the upload lets you move the set of files you are uploading into a specific hierarchy within the artifact without requiring you to move the files around on disk first.  This can be useful.